src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs

```diff
@@ -3650,6 +3650,35 @@ partial interface IGoo<T, U> { }
             Assert.Equal("T1", param.Name);
         }
 
+        [Fact]
+        public void PartialMethodRenameParameters()
+        {
+            var text = @"namespace NS
+{
+    public partial class MyClass
+    {
+        partial void F<T, U>(T t) where T : class;
+        partial void F<T, U>(T tt) where T : class {}
+    }
+}";
+            var comp = CreateCompilation(text);
+            comp.VerifyDiagnostics(
+                // (6,22): warning CS8826: Partial method declarations 'void MyClass.F<T, U>(T t)' and 'void MyClass.F<T, U>(T tt)' have signature differences.
+                //         partial void F<T, U>(T tt) where T : class {}
+                Diagnostic(ErrorCode.WRN_PartialMethodTypeDifference, "F").WithArguments("void MyClass.F<T, U>(T t)", "void MyClass.F<T, U>(T tt)").WithLocation(6, 22)
+                );
+
+            var ns = comp.SourceModule.GlobalNamespace.GetMembers("NS").Single() as NamespaceSymbol;
+            var type1 = ns.GetTypeMembers("MyClass").Single() as NamedTypeSymbol;
+            Assert.Equal(0, type1.TypeParameters.Length);
+            var f = type1.GetMembers("F").Single() as MethodSymbol;
+            Assert.Equal(2, f.TypeParameters.Length);
+            var param1 = f.TypeParameters[0];
+            var param2 = f.TypeParameters[1];
+            Assert.Equal("T", param1.Name);
+            Assert.Equal("U", param2.Name);
```

Comment:
Consider checking the parameter name rather than type parameter names since it is the parameter name that is distinct.

```suggestion
            Assert.Equal("T t", f.Parameters[0].ToTestDisplayString());;
```